Tooler/Controller/ShellController.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  ShellCommandExecutor.swift
+//  Tooler
+//
+//  Created by akash kahalkar on 27/12/23.
+//
+
+import Foundation
+
+protocol ShellProvider {
+    func safeShell(_ command: String, outputHandle: FileHandle?) async -> (output: String?, status: Int32)
+}
+
+final class DefaultShellController: ObservableObject, ShellProvider {
+    enum ShellError: Error {
+        case decodingError
+    }
+    
+    func safeShell(_ command: String, outputHandle: FileHandle? = nil) async -> (output: String?, status: Int32) {
```

Comment:
check this implementation by eskimo
https://developer.apple.com/forums/thread/690310